### PR TITLE
Fix invalid newline inside inline-table

### LINKED ISSUE #11: `pyproject-fmt` adding new line to inline-table
Hello, it seems that `pyproject-fmt` attempts to add an empty line to inline tables, which results in an invalid document.

For example, given the following file

```toml
[build-system]
build-backend = "setuptools.build_meta"
requires = ["setuptools"]

[project]
name = "myproj"
version = "0.4.2"
urls = {Homepage = "http://github.com"}
```

When I run the latest version of `pyproject-fmt`:
```bash
pipx run --spec 'pyproject-fmt>=0.3.1' pyproject-fmt pyproject.toml
```

I get the following (broken) TOML:
```toml
[build-system]
build-backend = "setuptools.build_meta"
requires = [
  "setuptools",
]

[project]
name = "myproj"
version = "0.4.2"
urls = {Homepage = "http://github.com"
}
```

As per the [spec](https://toml.io/en/v1.0.0#inline-table) inline tables should not span multiple lines[^1].

I can verify this TOML is broken by doing:
```pycon
$ pipx run --spec tomli python
Python 3.8.10 (default, Nov 26 2021, 20:14:08)
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import tomli
>>> from pathlib import Path
>>> tomli.loads(Path('pyproject.toml').read_text(encoding='utf-8'))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "~/.local/pipx/.cache/44b67bb1d3049b4/lib/python3.8/site-packages/tomli/_parser.py", line 102, in loads
    pos = key_value_rule(src, pos, out, header, parse_float)
  File "~/.local/pipx/.cache/44b67bb1d3049b4/lib/python3.8/site-packages/tomli/_parser.py", line 326, in key_value_rule
    pos, key, value = parse_key_value_pair(src, pos, parse_float)
  File "~/.local/pipx/.cache/44b67bb1d3049b4/lib/python3.8/site-packages/tomli/_parser.py", line 369, in parse_key_value_pair
    pos, value = parse_value(src, pos, parse_float)
  File "~/.local/pipx/.cache/44b67bb1d3049b4/lib/python3.8/site-packages/tomli/_parser.py", line 620, in parse_value
    return parse_inline_table(src, pos, parse_float)
  File "~/.local/pipx/.cache/44b67bb1d3049b4/lib/python3.8/site-packages/tomli/_parser.py", line 461, in parse_inline_table
    raise suffixed_err(src, pos, "Unclosed inline table")
tomli.TOMLDecodeError: Unclosed inline table (at line 10, column 39)
```

I suspect the problem might be coming from this line: https://github.com/tox-dev/pyproject-fmt/blob/3d6cd421dccdd356a7bacb5fabe02fcae5aca2d3/src/pyproject_fmt/formatter/util.py#L42

[^1]: There are a few exceptions here, e.g. a multi-line array or multi-line string inside of an inline table, but those are corner cases...

### PR DESCRIPTION
**Pull request by abravalheri**
Hello, this is a possible approach to fix #11.

The approach I took here was to pass the entire table object to the `order_keys` function, so it can be inspected to determine when it is appropriate (or not) to add an empty line.

I understand that the changes here might not be exactly what you have in mind, so please feel free to close this PR or ask for changes, I will be happy to implement them.

### CODE REVIEW

File: src/pyproject_fmt/formatter/project.py [hunk 1]
```diff
@@ -30,18 +30,18 @@ def fmt_project(parsed: TOMLDocument, conf: Config) -> None:
         opt_deps = cast(Table, project["optional-dependencies"])
         for value in opt_deps.values():
             normalize_pep508_array(cast(Array, value), conf.indent)
-        order_keys(opt_deps.value.body, (), sort_key=lambda k: k[0])
+        order_keys(opt_deps, (), sort_key=lambda k: k[0])
 
     for of_type in ("scripts", "gui-scripts", "entry-points", "urls"):
         if of_type in project:
             table = cast(Table, project[of_type])
```
Comment (abravalheri) [inline src/pyproject_fmt/formatter/project.py:37]: On a completely side-note here, you can also use `AbstractTable` in the places you are not sure if you are handling Table or InlineTable.

File: tests/formatter/test_project.py [hunk 2]
```diff
@@ -101,9 +101,7 @@ def test_entry_points(fmt: Fmt) -> None:
     """
     expected = """
     [project.entry-points]
-    alpha = {"A.A" = "a",B = "b"
-    }
-    beta = {C = "c",D = "d"
-    }
+    alpha = {"A.A" = "a",B = "b"}
```
Comment (gaborbernat) [inline tests/formatter/test_project.py:104]: This reads badly, can we add a space between `,` for inline tables? 🤔 while we're at it.

Comment (abravalheri) [reply tests/formatter/test_project.py:104]: I agree, I was about to ask if that was an stylistic decision, but I suppose it was not 😄 

I can give it a try, but currently `tomlkit` does not support that. So any solution I propose might sound hacky (I am afraid).

Comment (abravalheri) [reply tests/formatter/test_project.py:104]: @gaborbernat, I gave it a try in https://github.com/abravalheri/pyproject-fmt/pull/2, but the outcome was not very good.

Indeed, when I solve this problem, I introduce a different one.

I think the best course of action here is to add the sorting capability directly in `tomlkit`, then proceed doing the change here.

Comment (gaborbernat) [reply tests/formatter/test_project.py:104]: We'll have to follow-up on that then, let's accept this as is, one bugfix at a time.

**Pull request merged**